### ISSUE TITLE
Make the map on single locations better

### ISSUE THREAD

**Issue by arnaudbroes** [semperfiwebdesign/simplemap#12]
Make the map on single locations better

**Comment by wpsmort** [comment]
Make the map clickable (see issue #27)

**Comment by wpsmort** [comment]
Suggested code from Dave Iltis:

In templates.php, change to a larger map and add the website URL.  Code to fix is below. Change 100 to 400 and add the website URL...

```
// Grab the post that contains the template strucutre or the hard_coded structure
                       if ( 0 != $this->template_id ) {
                                // get post obejct via ID
                               // return post_content
                  } else {
                                $return = "<div class='sm-single-location-default-template'>";
                          $return .= "<div class='sm-single-map'>[sm-location data='iframe-map' map_width='400px' map_height='400px']<a href=\"[sm-location data='directions']\" target=\"_blank\"></a></div>";
                           $return .= "<div class='sm-single-location-data'>[sm-location data='full-address']";
                            $return .= "<br /><a href=\"[sm-location data='directions']\" target=\"_blank\">Get Directions</a>";
                            $return .= "<ul class='sm-single-location-data-ul'>[sm-location data='phone' before='<li>' after='</li>'] [sm-location data='email' before='<li><a href=\"mailto:%self%\">' after='</a></li>'] [sm-location data='sm_category' format='csv' before='<li>Categories: ' after='</li>'] [sm-location data='sm_tag' format='csv' before='<li>Tags: ' after='</li>']</ul>";
                         
/* added by Dave - this displays the URL on the individual shop page */
        $return .= "<br /><a href=\"[sm-location data='url']\">Visit Website</a>";
/* endadded by Dave */       

                                $return .= "</div>";
                            $return .= "<hr style='clear:both;' />";
                                $return .= "</div>";
                            $return .= "[sm-location data='description']";

                          return apply_filters( 'sm-single-location-default-template', $return );

                }

                       return $return;
        }
```

**Comment by wpsmort** [comment]
Currently we enforce the dimensions of the map on the single location page through a width and height specified in the iframe code. This makes it difficult for users to change using straight CSS.

Instead we should set the width and height of the iframe to 100% and then set the dimensions of the containing div so users can change this with CSS in their theme.

We first need to fix the bug with #141 and then I can add styling for this.

**Comment by wpsmort** [comment]
This map on the single location page is now responsive but in order to fully test this I need to resolve #168 first.

**Comment by wpsmort** [comment]
This is tested and is good.  I tested with Twenty Fifteen, Twenty Sixteen and Twenty Seventeen themes.

**Comment by wpsmort** [comment]
I've changed the ratio of the map and removed all other code that controls map width and height so that these dimensions are now controlled by CSS.  This is as a result of further responsive testing with other themes and page layouts.

For reference, I'm using the code provided at the link below for making Google Maps iFrames responsive:

https://www.smashingmagazine.com/2014/02/making-embedded-content-work-in-responsive-design/#responsive-google-maps

**Comment by michaeltorbert** [comment]
@wpsmort 
My advice is to make separate issues for each specific thing we want, instead of (or in addition to) just a general "make it better".

**Comment by wpsmort** [comment]
@michaeltorbert This issue is only for changing the size of the map on single locations and for linking it to Google Maps (the linking was added in version 2.4.9).  These changes were suggested by Dave Iltis.

The result of changes done for the next release is that the map is now responsive and we no longer constrain or set the dimensions.  The map dimensions can now be controlled with a line of CSS and I've added an FAQ about this here - http://simplemap-plugin.com/faqs/

Other improvements to the map for this release have been made under #170.

I've now tested the new responsive map with TwentyFourteen and onwards, as well as ten of the most popular themes on WP.org and with Genesis.  The map responds well on all screen sizes.  

This issue can now be closed.
